e2e/containers.e2e.spec.ts

```diff
@@ -0,0 +1,106 @@
+import os from 'os';
+import path from 'path';
+
+import { test, expect } from '@playwright/test';
+import { ElectronApplication, BrowserContext, _electron, Page } from 'playwright';
+
+import { NavPage } from './pages/nav-page';
+import { createDefaultSettings, packageLogs, reportAsset, tool } from './utils/TestUtils';
+
+import { ContainerEngine, Settings } from '@/config/settings';
+import { RecursivePartial } from '~/utils/typeUtils';
+
+let page: Page;
+
+/**
+ * Using test.describe.serial make the test execute step by step, as described on each `test()` order
+ * Playwright executes test in parallel by default and it will not work for our app backend loading process.
+ * */
+test.describe.serial('Containers Test', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item as the final option.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ],
+      env: {
+        ...process.env,
+        RD_LOGS_DIR: reportAsset(__filename, 'log'),
+      },
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: reportAsset(__filename) });
+    await packageLogs(__filename);
+    await electronApp.close();
+  });
+
+  test('wait for the container engine to be ready', async() => {
+    const navPage = new NavPage(page);
+
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBar).toBeHidden();
+  });
+
+  test('should ensure sudo is disabled', async() => {
```

Comment:
This can be skipped by just passing in the override options to `createDefaultSettings` on line 24.  That reduces a backend restart, so should run faster.

Reply:
Good catch. I didn't know that function could be parameterized.